scripts/oec_data_source_script.rb

```diff
@@ -0,0 +1,154 @@
+    sleep 2
+    @driver.find_element(id: 'AdminUC_DataSources_AdminDataSource_Tabs_tbSearchValue').send_keys data_source
+    @driver.find_element(id: 'AdminUC_DataSources_AdminDataSource_Tabs_btnSearch').click
+    @brief_wait.until { @driver.find_element(xpath: "//table[@id='AdminUC_DataSources_AdminDataSource_Tabs_MultiDataSource_listing']//tr[3][contains(.,'#{data_source}')]") }
+    @driver.find_element(link_text: 'Edit').click
+    # Click Data tab and click Edit on Data Blocks tab
+    puts "#{Time.now} - Found data source, editing it"
+    click_element_id(@driver, 'AdminUC_Data_Data')
+    @brief_wait.until do
+      @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_lblDataSource')
+      @brief_wait.until { @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_lblDataSource').text.include? data_source }
+    end
+    @driver.find_element(link_text: 'Edit').click
+  end
+
+  def upload_file(file)
+    puts "#{Time.now} - Uploading file '#{file}'"
+    @brief_wait.until { @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_File1') }
+    @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_File1').send_keys file
+    @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_btnUpload').click
+    @long_wait.until { @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_cbAll') }
+  end
+
+  def connect_source
+    puts "#{Time.now} - Connecting data source"
+    click_element_id(@driver, 'AdminUC_Data_ucAdminDS_Entities_btnUpload')
+    @brief_wait.until { @driver.find_element(id: 'AdminUC_Data_ucAdminDS_Entities_cbAll') }
+  end
+
+  def apply_and_import_source
+    # Apply data block update
+    puts "#{Time.now} - Applying and importing data source"
+    click_element_id(@driver, 'AdminUC_Data_ucAdminDS_Entities_cbAll')
+    sleep 2
+    @brief_wait.until do
+      @driver.find_elements(xpath: '//table[@id="AdminUC_Data_ucAdminDS_Entities_FieldTable"]//input[@type="checkbox"]').each { |e| e.selected? }
+    end
+    click_element_id(@driver, 'AdminUC_Data_ucAdminDS_Entities_btnAdd')
+    @long_wait.until { @driver.find_element(xpath: '//a[contains(.,"Data Block has been updated successfully.")]') }
+    sleep 2
+    # Import data source
+    click_element_id(@driver, 'AdminUC_Data_Import/Export')
+    click_element_id(@driver, 'AdminUC_Data_AdminDS_Import_btnImport')
+    @long_wait.until { @driver.find_element(xpath: '//span[contains(.,"Data Import Successful")]') }
+    click_element_id(@driver, 'AdminUC_Data_AdminDS_Import_btnConfirm')
+    @long_wait.until { @driver.find_element(xpath: '//span[contains(.,"Data Import Approved and Successful")]') }
+    @brief_wait.until { @driver.find_element(id: 'AdminUC_Data_btnSave') }
+    puts "#{Time.now} - Import succeeded"
+  end
+
+  def create_results_file
+    output_dir = File.join(ENV['HOME'], 'selenium-files')
+    output_file = 'data_source_update_results.log'
```

Comment:
No @johncrossman, since each run overwrites the file, the timestamp in the file system should be enough to know when the run occurred.